core/modules/macros/contrastcolour.js

```diff
@@ -29,6 +29,22 @@ exports.params = [
 Run the macro
 */
 exports.run = function(target,fallbackTarget,colourA,colourB) {
```

Comment:
I don't think coupling contrastcolour to palettes in general and specific palette entries is necessary or desirable. The existing implementation avoids it, why can't we do so here?